pkg/packetfilter/nftables/namedset.go

```diff
@@ -0,0 +1,142 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nftables
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner/pkg/packetfilter"
+	"sigs.k8s.io/knftables"
+)
+
+type namedSet struct {
+	set      knftables.Set
+	nftables knftables.Interface
+}
+
+func (p *packetFilter) NewNamedSet(set *packetfilter.SetInfo) packetfilter.NamedSet {
+	setType := "ipv4_addr"
+
+	return &namedSet{
+		set: knftables.Set{
+			Name: set.Name,
+			Type: setType,
+		},
+		nftables: p.nftables,
+	}
+}
+
+func (n *namedSet) Create(_ bool) error {
+	tx := n.nftables.NewTransaction()
+
+	tx.Add(&n.set)
+	err := n.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrap(err, "error creating set")
+}
+
+func (n *namedSet) Name() string {
+	return n.set.Name
+}
+
+func (n *namedSet) Flush() error {
+	tx := n.nftables.NewTransaction()
+
+	tx.Flush(&n.set)
+	err := n.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrap(err, "error flushing set")
+}
+
+func (n *namedSet) Destroy() error {
+	tx := n.nftables.NewTransaction()
+	tx.Delete(&n.set)
+
+	err := n.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrap(err, "error deleting set")
+}
+
+func (n *namedSet) AddEntry(entry string, _ bool) error {
+	tx := n.nftables.NewTransaction()
+
+	tx.Add(&knftables.Element{
+		Set: n.set.Name,
+		Key: []string{entry},
+	})
+
+	err := n.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrap(err, "error adding entry to set")
+}
+
+func (n *namedSet) DelEntry(entry string) error {
+	tx := n.nftables.NewTransaction()
+
+	tx.Delete(&knftables.Element{
+		Set: n.set.Name,
+		Key: []string{entry},
+	})
+
+	err := n.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrap(err, "error deleting entry to set")
+}
+
+func (n *namedSet) ListEntries() ([]string, error) {
+	nftablesElems, err := n.nftables.ListElements(context.TODO(), "set", n.set.Name)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "error listings elements")
+	}
+
+	strElems := []string{}
+
+	for _, existingElem := range nftablesElems {
+		strElems = append(strElems, existingElem.Key[0])
+	}
+
+	return strElems, nil
+}
+
+func (p *packetFilter) DestroySets(nameFilter func(string) bool) error {
+	setsList, err := p.nftables.List(context.TODO(), "sets")
+	if err != nil && !knftables.IsNotFound(err) {
+		return errors.Wrap(err, "error listing sets")
+	}
+
+	retErr := error(nil)
+
+	for _, set := range setsList {
+		if nameFilter(set) {
+			tx := p.nftables.NewTransaction()
+			tx.Delete(&knftables.Set{
+				Name: set,
+			})
+
+			err = p.nftables.Run(context.TODO(), tx)
+			if err != nil && !knftables.IsNotFound(err) {
+				logger.Errorf(err, "Error destroying the set %q", set)
+				retErr = err
+			}
+		}
+	}
+
+	return retErr
```

Comment:
To avoid code duplication, we can extract a function to delete a set:

```
func deleteSet(nftables knftables.Interface, name string) error {
	tx := nftables.NewTransaction()
	tx.Delete(&knftables.Set{
		Name: name,
	})

	err := nftables.Run(context.TODO(), tx)
	if knftables.IsNotFound(err) {
		return nil
	}

	return errors.Wrapf(err, "error deleting set %q", name)
}
```

which we can also call from `namedSet.Destroy().`. Also we should aggregate the errors here so we don't lose any in case there's multiple

```suggestion
import utilerrors "k8s.io/apimachinery/pkg/util/errors"
...

	var errs []error

	for _, set := range setsList {
		if nameFilter(set) {
			err = deleteSet(p.nftables, set)
			errs = append(errs, err)
		}
	}

	return utilerrors.NewAggregate(errs)
```